core/src/main/java/com/cryptomorin/xseries/XGameRule.java

```diff
@@ -139,15 +143,61 @@ public enum XGameRule implements XBase<XGameRule, String> {
     @XInfo(since = "?", removedSince = "1.21.11")
     DO_FIRE_TICK("doFireTick");
 
+    private static final boolean SUPPORTS_GAME_RULE_API;
+    private static final MethodHandle GET_BY_NAME;
+    private static final MethodHandle GET_GAME_RULE_VALUE;
+    private static final MethodHandle SET_GAME_RULE_VALUE;
+
+    static {
+        boolean supportsGameRuleAPI = true;
+        MethodHandle getByName = null;
+        MethodHandle getGameRuleValue = null;
+        MethodHandle setGameRuleValue = null;
+
+        MethodHandles.Lookup methodHandles = MethodHandles.lookup();
+        try {
+            getByName = methodHandles.unreflect(GameRule.class.getDeclaredMethod("getByName", String.class));
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } catch (NoClassDefFoundError e) {
+            supportsGameRuleAPI = false;
+
+            try {
+                getGameRuleValue = methodHandles.unreflect(World.class.getDeclaredMethod("getGameRuleValue", String.class));
+                setGameRuleValue = methodHandles.unreflect(World.class.getDeclaredMethod("setGameRuleValue", String.class, String.class));
```

Comment:
Can't we just use the lookup directly? Why have the reflection intermediate?

```java
lookup.findVirtual(World.class, "setGameRuleValue", MethodType.methodType(void.class, String.class, String.class));
```

It really doesn't really matter, with direct findVirtual lookups the startup performance and memory usage might improve which is negligible, it's mostly a matter of convention. I'd appreciate if you can fix the other 2 as well.